### PR TITLE
Parse megawatcher response

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

File: jujugui/static/gui/src/app/maraca/maraca.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
+/* Copyright (C) 2018 Canonical Ltd. */
+'use strict';
+
+const clone = require('lodash.clone');
+const deepmerge = require('deepmerge');
+
+const parsers = require('./parsers');
+
+class Maraca {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/maraca/maraca.js]: Bikeshed, begin!

Comment (hatched) [reply jujugui/static/gui/src/app/maraca/maraca.js]: 🚲

File: jujugui/static/gui/src/app/init.js [hunk 2]
```diff
@@ -263,6 +264,14 @@ class GUIApp {
       }
     });
 
+    this.maraca = new Maraca({
+      changeEvent: '_rpc_response',
+      onChange: () => {
+        this._renderApp(this.state.current, () => {});
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply jujugui/static/gui/src/app/init.js]: I haven't passed it yet, I'll do that in a followup when I update the status components. It'll be done like this:
```
<App
    ...
    valueStore={this.maraca.values}
    ... />
```

File: jujugui/static/gui/src/app/init.js [hunk 3]
```diff
@@ -263,6 +264,14 @@ class GUIApp {
       }
     });
 
+    this.maraca = new Maraca({
+      changeEvent: '_rpc_response',
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply jujugui/static/gui/src/app/init.js]: Done.

File: jujugui/static/gui/src/app/maraca/maraca.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    }
+  }
+
+  /**
+    Parse the entity response into a friendly format.
+    @param entityType {String} The type of entity.
+    @param entity {Object} The entity details.
+    @returns {Object} The parsed entity.
+  */
+  _parseEntity(entityType, entity) {
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply jujugui/static/gui/src/app/maraca/maraca.js]: Nope, it's part of the delta e.g. `['unit', 'change', {...details...}]`.

File: jujugui/static/gui/src/app/init.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (hatched) [reply jujugui/static/gui/src/app/init.js]: I suppose you'd want a `this.maraca.getValues()` which returns a deep frozen object.

[the remark this answers is not in the document]
Comment (huwshimi) [reply jujugui/static/gui/src/app/init.js]: Done.

File: jujugui/static/gui/src/app/maraca/test-parsers.js [hunk 5]
```diff
@@ -0,0 +1,3151 @@
...
+
+
+const parsers = require('./parsers');
+
+describe('parse entities', () => {
+  let parsed, response;
+
+  beforeEach(() => {
+    parsed = {
+      annotations: {
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply jujugui/static/gui/src/app/maraca/test-parsers.js]: Done.

**Pull request merged**